core/domain/skill_jobs_one_off_test.py

```diff
@@ -0,0 +1,103 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for Skill-related one-off jobs."""
+
+from core.domain import skill_domain
+from core.domain import skill_jobs_one_off
+from core.domain import skill_services
+from core.platform import models
+from core.tests import test_utils
+import feconf
+
+(skill_models,) = models.Registry.import_models([models.NAMES.skill])
+
+
+class SkillMigrationJobTest(test_utils.GenericTestBase):
+
+    ALBERT_EMAIL = 'albert@example.com'
+    ALBERT_NAME = 'albert'
+
+    SKILL_ID = 'skill_id'
+
+    def setUp(self):
+        super(SkillMigrationJobTest, self).setUp()
+
+        # Setup user who will own the test skills.
+        self.albert_id = self.get_user_id_from_email(self.ALBERT_EMAIL)
+        self.signup(self.ALBERT_EMAIL, self.ALBERT_NAME)
+        self.process_and_flush_pending_tasks()
+
+    def test_migration_job_does_not_convert_up_to_date_skill(self):
```

Comment:
Ditto re tests that ensure v1 is converted to vN.

Also try and find ways to test situations where (i) both schemas have updates, (ii) one schema has updates but the other does not.

Reply:
Done. Note that I haven't added actual tests that do this yet, but with the `test_util` function in place, it is just a single call change in the existing tests to implement this. I think they should only be added depending on the future version changes.